src/plugins/intel_gna/src/transformations/insert_copy_layer.cpp

```diff
@@ -43,6 +47,48 @@ namespace {
 
         output_op->input(index).replace_source_output(copy_op);
     }
+
+    /**
+     * Return true in case Copy Layer should be inserted. Two conditions must be met:
+     * - input of Brocast layer is also the input layer of network ( skipping non-functinal).
+     * - Broadcast will be finally removed from network - In case product of input and output shape are the same.
+     */
+    bool ShouldInsertCopyBeforeBroadcast(std::shared_ptr<ov::Node>& broadcast_input,
+                                         std::shared_ptr<Broadcast>& broadcast) {
+        // skip non functional layers
+        auto current_node = get_prev_node_skipping_certain(broadcast_input, is_gna_non_functional_node);
+
+        if (!std::dynamic_pointer_cast<Parameter>(current_node)) {
+            return false;
+        }
+
+        auto data_node = broadcast->input_value(0);
+
+        static constexpr size_t index_of_target_shape = 1;
+
+        auto target_shape_node =
+            std::dynamic_pointer_cast<Constant>(broadcast->input_value(index_of_target_shape).get_node_shared_ptr());
+        if (!target_shape_node)
+            return false;
+
+        auto input_shape = data_node.get_shape();
+        auto output_shape = ov::Shape(target_shape_node->cast_vector<size_t>());
+
+        // if input_shape and output_shape ranks are different
+        if (input_shape.size() != output_shape.size()) {
+            // if input shape rank is higher than output shape rank return false;
+            if (input_shape.size() > output_shape.size()) {
+                return false;
+            }
+        }
```

Comment:
Basically it is the same as:
if (input_shape.size() > output_shape.size()) return false;